modules/update-duration-cvo.adoc

```diff
@@ -12,3 +12,26 @@ The Cluster Version Operator (CVO) retrieves the target update release image and
 ====
 The CVO phase of the update does not restart the nodes.
 ====
+
+[discrete]
+== Example update duration of cluster Operators
+
+// image to be replaced with a properly designed version
+image::update-duration.png[A diagram displaying the periods during which cluster Operators report as progressing during an update]
+
+The previous diagram shows an example of the time that cluster Operators might take to update to their new versions.
+The bars on the graph represent periods during which the respective Operator reports a `Progressing=True` condition.
+During an update, the `Progressing=True` condition is triggered mostly when the Operator is updating itself, but for some Operators, other tasks related to the update may trigger the condition as well.
+
+Each cluster Operator has characteristics that affect the time it takes to update itself.
+For instance, the `kube-apiserver` Operator in this example took over nine minutes to update because it was serving customer requests and is designed to be conservative about draining out existing requests before shutting down a pod.
+In the case of the `kube-apiserver` Operator, update speed is sacrificed to help prevent and limit disruptions to cluster functionality during an update.
+
+Another characteristic that affects the update duration of an Operator is whether the Operator utilizes DaemonSets.
+The `network` and `dns` Operators utilize full-cluster DaemonSets, which can take time to roll, and this is one of several reasons why these Operators might take longer to update themselves.
+// To do: request clarification on how "roll" could be better described in a way that makes sense to users. What is being done to/with an Operator's DaemonSets during an update?
+
+[NOTE]
+====
+Each cluster Operator waits until its designated stage, or Runlevel, to begin its update. Operators within the same Runlevel can update simultaneously. For more information about Runlevels, see "Understanding how manifests are applied during an update" in the "Additional resources" section.
```

Comment:
@wking this is accurate, right? For the purpose of explaining bars in the graph that are vertically overlapping, is this good enough, or do I need to describe things in terms of manifest application?

Reply:
nit: I'm not sure we want to make "Runlevel" a formal word.  Maybe we can keep "run level" as an informal local phrase to avoid having customers think that this is something they need to internalize for future use outside of trying to understand update duration?

> Each cluster Operator waits until its designated stage, or Runlevel, to begin its update.

The effect is correct, but the current phrasing suggests it is a component-operator decision.  In reality, it's the cluster-version operator waiting until previous run-levels have been reconciled before updating a cluster operator's Deployment.  That incoming cluster operator pod launches, realizes that it's vB while the deployed component is mostly vA, and proceeds to update its component operands until they've reached vB.  So the actor enforcing the runlevel ordering is the cluster-version operator, not the individual component operators.